src/components/Comment/CommentSection/CommentSection.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { ReactNode, useRef, useState } from 'react';
+import { CommentData } from 'src/types/domain';
+
+import CommentGroup from '../CommentGroup';
+import Comment from '../Comment';
+import { CommentSectionWrapper, NewCommentWrapper } from './styled';
+import TextAreaField from 'src/components/fields/TextAreaField';
+import Button from 'src/components/inputs/Button';
+import ScrollableContainer from 'src/components/ScrollableContainer';
+import ReplyingToIndicator from 'src/components/Comment/ReplyingToIndicator';
+
+import { createComment } from 'src/api/comments';
+
+type CommentSectionProps = {
+  eventId: string;
+  comments: Array<CommentData>;
+  refreshEvent: Function;
+};
+
+export default function CommentSection({
+  eventId,
+  comments,
+  refreshEvent,
+}: CommentSectionProps) {
+  const [commentBody, setCommentBody] = useState('');
+  const [replyToCommentId, setReplyToCommentId] = useState<string | undefined>(
+    undefined
+  );
+  const [replyToCommentAuthor, setReplyToCommentAuthor] = useState<
+    string | undefined
+  >(undefined);
+
+  const textAreaRef = useRef<HTMLTextAreaElement>(null);
+
+  const setTextAreaFocus = () => {
+    textAreaRef.current?.focus();
+  };
+
+  const topLevelComments = getTopLevelComments(comments);
+  const commentNodes = generateCommentNodes(
+    topLevelComments,
+    setReplyToCommentId,
+    setReplyToCommentAuthor,
+    setTextAreaFocus
+  );
+
+  const handleSubmitComment = async (
+    eventId: string,
+    commentBody: string,
+    replyToCommentId: string | undefined
+  ) => {
+    await createComment(eventId, commentBody, replyToCommentId);
+    refreshEvent();
+    setCommentBody('');
+    setReplyToCommentAuthor(undefined);
+    setReplyToCommentId(undefined);
```

Comment:
Sjekk at man ikke skriver tomme kommentarer.
Gi en tilbakemelding til bruker, kanskje ikke en alert, men en liten melding :)
```suggestion
    if (commentBody === "") alert("There is no need to add an empty comment :p");
    else {
      await createComment(eventId, commentBody, replyToCommentId);
      refreshEvent();
      setCommentBody('');
      setReplyToCommentAuthor(undefined);
      setReplyToCommentId(undefined);
    }
```

Reply:
Merk at det også går an å bare skrive `if (!commentBody)` her 😛